pkg/hooks/manager.go

```diff
@@ -103,22 +103,38 @@ func (m *hookManager) collectSideCarSockets(numberOfRequestedHookSidecars uint,
 	timeoutCh := time.After(timeout)
 
 	for uint(len(processedSockets)) < numberOfRequestedHookSidecars {
-		sockets, err := os.ReadDir(m.hookSocketSharedDirectory)
+		entries, err := os.ReadDir(m.hookSocketSharedDirectory)
 		if err != nil {
 			return nil, err
 		}
 
-		for _, socket := range sockets {
-			if _, processed := processedSockets[socket.Name()]; processed {
+		for _, entry := range entries {
+			if !entry.IsDir() {
 				continue
 			}
 
-			err := handleSidecarSocket(filepath.Join(m.hookSocketSharedDirectory, socket.Name()), callbacksPerHookPoint)
+			subPath := filepath.Join(m.hookSocketSharedDirectory, entry.Name())
+			subEntries, err := os.ReadDir(subPath)
 			if err != nil {
 				return nil, err
 			}
 
-			processedSockets[socket.Name()] = true
+			for _, subEntry := range subEntries {
+				if subEntry.IsDir() {
```

Comment:
Please consider also skipping non Unix socket files.

Reply:
I prefer not in this PR, we can always do follow-ups upon needs,
it wasn't there before, and we can't encourage putting other files in this folder, as then someone can put non gRPC socket (fwiw also before this PR one couldn't put there other files)